### ISSUE TITLE
Link dialogue appears top left when there is no anchor 

### ISSUE THREAD

**Issue by fredachang** [mdx-editor/editor#175]
Hi petyosi, 

Thank you for this editor - really enjoyed working with it as it was very easy to import and get started right away. 

One issue we are facing that we can't seem to fix ourselves is the fact that the link dialogue defaults to the top left of the screen when there is anchor i.e. when cursor is on an empty line or an empty bullet point. I can see the radix styling on the parent div of the dialogue but there is no way to override that with custom css as we cannot target that div. 

Is it possible to either 

- have a class name on the parent div so that we can target with our own css?
- have the default position of the link dialogue in the middle of the page when no anchor / text are selected?

Much appreciated, 
Freda

**Comment by fredachang** [comment]
Thanks for the quick response! I can see the default position issue has been fixed and is definitely a workable solution for us. 

However there are some unintended side effects with this upgrade. Previously if the cursor is on a sentence or paragraph (with no text selected), it would just link the entire sentence/paragraph which we didn't mind. But now it would just insert the url in like below:

![Screenshot 2023-11-14 at 10 23 54 am](https://github.com/mdx-editor/editor/assets/128881398/47b2eb47-3dfc-4be9-a230-ace4f9cc9d2f)

In short, it would be great if:
- it pastes the url text in if there is no text before or after the cursor (e.g. on empty line / empty bullet point) like the current implementation 
- but if there is text before or after the cursor, use that as the title

Thanks again, and totally understand if this level of granular control is not possible.

**Comment by fredachang** [comment]
Another simpler solution I thought of is just to disable to link button when no text is selected. Would that be a complex implementation or cause unexpected results? happy to have a go at this and contribute.

**Comment by petyosi** [comment]
That's quite simple to implement. Following is some untested code. But I'm not sure if letting users insert a link without selection is not a good feature. 

```tsx
const [selection] = useCoreHooks('currentSelection`)
...
disabled={!selection.isRangeSelection() selection.isCollapsed)
```

**Comment by fredachang** [comment]
Awesome. Agree the disabled feature might be more specific to my use case than general. I'll have a think about the suggested code but will probably want to stay away from forking our own branch. 

For context - we've actually abstracted the title field away and modified the link dialogue to below as our users will almost always link existing content directly (a more similar experience to notion). 
<img width="388" alt="Screenshot 2023-11-14 at 9 10 52 pm" src="https://github.com/mdx-editor/editor/assets/128881398/f9e85582-780b-4584-b7e3-8c611894afe5">
